bolt-modules/boltlib/lib/puppet/functions/apply_prep.rb

```diff
@@ -38,20 +38,24 @@ def apply_prep(target_spec)
 
     executor.log_action('install puppet and gather facts', targets) do
       executor.without_default_logging do
-        # Ensure Puppet is installed
-        versions = run_task(executor, targets, 'puppet_agent::version')
-        need_install, installed = versions.partition { |r| r['version'].nil? }
-        installed.each do |r|
-          Puppet.info "Puppet Agent #{r['version']} installed on #{r.target.name}"
-        end
+        # Skip targets run over PCP, as we know an agent will be available.
```

Comment:
I'm tempted to make this check whether the target's include the puppet-agent feature (either from inventory or transport). That would provide a way for people to use apply_prep and bypass agent setup if they know (or we can detect externally) that the agent is available.